src/overlays/actors/ovl_En_Sth/z_en_sth.c

```diff
@@ -169,142 +198,144 @@ void EnSth_Destroy(Actor* thisx, PlayState* play) {
     Collider_DestroyCylinder(play, &this->collider);
 }
 
-s32 func_80B6703C(EnSth* this, PlayState* play) {
+s32 EnSth_DetectSpeakingRange(EnSth* this, PlayState* play) {
     if ((this->actor.xzDistToPlayer < 100.0f) && Player_IsFacingActor(&this->actor, 0x3000, play) &&
         Actor_IsFacingPlayer(&this->actor, 0x2000)) {
         return true;
+    } else {
+        return false;
     }
-    return false;
 }
 
-void func_80B670A4(EnSth* this, s16 arg1) {
-    if ((arg1 >= 0) && (arg1 < ARRAY_COUNT(D_80B6D1C8)) && (arg1 != this->unk_29A)) {
-        Animation_Change(&this->skelAnime, D_80B6D1C8[arg1], 1.0f, 0.0f, Animation_GetLastFrame(D_80B6D1C8[arg1]),
-                         ANIMMODE_LOOP, -5.0f);
-        this->unk_29A = arg1;
+void EnSth_SwitchAnimation(EnSth* this, s16 animIndex) {
+    if ((animIndex >= 0) && (animIndex < ARRAY_COUNT(sAnimationInfo)) && (animIndex != this->curAnimIndex)) {
+        Animation_Change(&this->skelAnime, sAnimationInfo[animIndex], 1.0f, 0.0f,
+                         Animation_GetLastFrame(sAnimationInfo[animIndex]), ANIMMODE_LOOP, -5.0f);
+        this->curAnimIndex = animIndex;
     }
 }
 
-void func_80B67148(EnSth* this, PlayState* play) {
+void EnSth_GetInitialDispairText(EnSth* this, PlayState* play) {
     s32 day = CURRENT_DAY - 1;
-    u16 val;
+    u16 nextTextId;
 
     if (day < 0) {
         day = 0;
     }
-    val = D_80B6D1E8[day];
+    nextTextId = sSthOceanspiderhouseDispairText[day];
 
-    Message_StartTextbox(play, val, &this->actor);
+    Message_StartTextbox(play, nextTextId, &this->actor);
 }
 
-void func_80B671A0(EnSth* this, PlayState* play) {
+void EnSth_HandleDispairConversation(EnSth* this, PlayState* play) {
     SkelAnime_Update(&this->skelAnime);
 
     if ((Message_GetState(&play->msgCtx) == TEXT_STATE_5) && Message_ShouldAdvance(play)) {
-        this->actionFunc = func_80B67208;
+        this->actionFunc = EnSth_DispairIdle;
         func_801477B4(play);
     }
 }
 
-void func_80B67208(EnSth* this, PlayState* play) {
+void EnSth_DispairIdle(EnSth* this, PlayState* play) {
     SkelAnime_Update(&this->skelAnime);
 
     if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
-        func_80B67148(this, play);
-        this->actionFunc = func_80B671A0;
+        EnSth_GetInitialDispairText(this, play);
+        this->actionFunc = EnSth_HandleDispairConversation;
     } else if ((this->actor.xzDistToPlayer < 100.0f) && Player_IsFacingActor(&this->actor, 0x3000, play)) {
         func_800B8614(&this->actor, play, 110.0f);
     }
 }
 
-void func_80B672A4(EnSth* this, PlayState* play) {
-    u16 sp1E;
+void EnSth_GetInitialOceanSpiderhouseText(EnSth* this, PlayState* play) {
+    u16 nextTextId;
     s32 day = CURRENT_DAY - 1;
 
     if (day < 0) {
         day = 0;
     }
 
-    if (this->unk_29C & 2) {
+    if (this->sthFlags & STH_FLAG_CURED) {
```

Comment:
This seems to be for the Oceanside Spider House, but the flag of "cured" doesn't really fit for this guy. Perhaps there's another flag that uses the same value?